### ISSUE TITLE
Related Posts: Customizing title in output

### ISSUE THREAD

**Issue by adamgamble** [Automattic/jetpack#3265]
So we're currently using the jetpack plugin to show related posts on our articles. This is working great but we would like the customize the html output pretty dramatically. We also would like to add in extra data to the api output to be used in the html.

Are there plans to create hooks to return more data in the api for a post? If not would you accept a PR providing this?

Are there plans to create a hook to customize html (or does it already exist) of related posts, if not would you accept a PR providing this?

Basically my use case is I want to change the title of the post to be a field in the post meta, and I want the html to only show the title of the post with custom styling.

Thanks for your help

**Comment by kraftbj** [comment]
One idea is to use the https://github.com/Automattic/jetpack/blob/master/modules/related-posts/jetpack-related-posts.php#L1266 Raw class. This is meant to get back results from the API without any front-end work, meaning you can build out anything you'd like.

That would return the post IDs only (all that the related posts API itself returns, the rest of the processing is done locally).

As far as what we'd accept for PRs on this, I defer to @xyu. If you'd like to open a PR for one/both or paste in a proof of concept to make sure we're on the same page, that would be helpful.

**Comment by adamgamble** [comment]
Ok I think i'll just build a custom plugin that uses your Raw class and build the functionality I need.

Thanks!

**Comment by kraftbj** [comment]
Sounds great. Cheers!
